ksql-api/src/main/java/io/confluent/ksql/api/server/QueryStreamHandler.java

```diff
@@ -80,10 +79,11 @@ public void handle(final RoutingContext routingContext) {
     createQueryPublisherAsync(queryStreamArgs.get().sql, queryStreamArgs.get().properties, context)
         .thenAccept(queryPublisher -> {
 
-          final PushQueryHolder query = connectionQueryManager
-              .createApiQuery(queryPublisher, routingContext.request());
+          final PushQueryHolder query = queryPublisher.isPullQuery() ? null :
+              connectionQueryManager.createApiQuery(queryPublisher, routingContext.request());
```

Comment:
Rather than having `null` here, why not introduce a `QueryHolder` interface with the `getId` and `close` methods. Then `connectionQueryManager.createApiQuery` can return a `PushQueryHolder` for push queries and a `PullQueryHolder` for pull. The Pull version would have a no-op `close` method.

This uses the power of polymorphism to avoids all the null checking and `if`s.

Reply:
+1 this would avoid (at a minimum) the two if checks below and at best future NPEs from less detailed oriented coders (like me)

Reply:
I think this is a matter of personal style. I think the current code is simple and straight to the point. Adding new types here seems overkill.

Reply:
This is not a matter of style. Style is about where you put your brackets or tabs vs spaces.  This is about good programming patterns and OO principles.  In this case its leveraging polymorphism to avoid conditionals.

Reply:
I'd consider trading a single succint line of code for a new interface, probably 20+ lines of new code and changes, as overengineering it.